ui/src/main/webapp/src/app/shared/filter/filter.pipe.ts

```diff
@@ -0,0 +1,21 @@
+import {Pipe, PipeTransform} from "@angular/core";
+
+@Pipe({
+    name: 'wuFilter',
+    pure: true
+})
+export class FilterPipe implements PipeTransform {
+
+    transform(array: any[], filter: (item: any) => boolean): any {
+        console.log('FilterPipe called', {
+            data: array,
+            params: filter
+        });
+
+        if (!filter) {
+            return array;
+        }
+
+        return array.filter(filter);
+    }
+}
```

Comment:
We could get rid of this if we pass the filter directly to the data table component. I mean if the filtering is handled in the TS code rather than moving the pipe to the component's template.

Reply:
This could be used on multiple places where you need filtering. Not only for data table.

Reply:
Sure, let's keep that